src/non_zero.rs

```diff
@@ -51,20 +51,22 @@ impl<T> NonZero<T> {
         CtOption::new(Self(n), !is_zero)
     }
 
-    /// Provides access to the contents of `NonZero` in a `const` context.
-    pub const fn as_ref(&self) -> &T {
-        &self.0
-    }
-
     /// Returns the inner value.
     pub fn get(self) -> T {
         self.0
     }
 }
 
+impl<T: ?Sized> NonZero<T> {
+    /// Provides access to the contents of `NonZero` in a `const` context.
+    pub const fn as_ref(&self) -> &T {
```

Comment:
I've been wondering if we should use a different method name for these which doesn't clash with the `AsRef` trait, but this is at least consistent with `Odd`